AnnotatoBackend/Sources/App/WebSocketControllers/AnnotationWebSocketController.swift

```diff
@@ -110,6 +112,37 @@ class AnnotationWebSocketController {
         }
     }
 
+    private func handleCreateOrUpdateAnnotation(
+        userId: String,
+        db: Database,
+        annotation: Annotation
+    ) async -> Annotation? {
+        do {
+            self.logger.info("Processing create or update annotation... \(String(describing: annotation))")
+            let createdOrUpdatedAnnotation: Annotation
+
+            if await annotationDataAccess.canFindWithDeleted(db: db, annotationId: annotation.id) {
+                createdOrUpdatedAnnotation = try await annotationDataAccess.update(
```

Comment:
Quick question, if the annotation was updated, why can't it just send an `.updateAnnotation` message back? And if it was created, why can't it just send a `.createAnnotation` message back? Do we need a `.createOrUpdateAnnotation` message?

Reply:
Yup I agree we need a diff message when sending.

Also about the last point, when the local has it, it is stored as the original id, but when it's rendered as a model/view model, it's given a new id isn't it? So does that mean the local persistence does not have it yet?

Reply:
Ok can